projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -117,25 +173,25 @@ public function action_admin_init() {
 	 * @global $_GET
 	 * @action wp
 	 * @uses add_shortcode, get_the_ID
-	 * @returns null
 	 */
 	public function action_frontend_init() {
-		// Add a shortcode handler that outputs nothing, this gets overridden later if we can display related content
+		// Add a shortcode handler that outputs nothing, this gets overridden later if we can display related content.
 		add_shortcode( self::SHORTCODE, array( $this, 'get_client_rendered_html_unsupported' ) );
 
-		if ( ! $this->_enabled_for_request() )
+		if ( ! $this->enabled_for_request() ) {
 			return;
+		}
 
 		if ( isset( $_GET['relatedposts'] ) ) {
 			$excludes = $this->parse_numeric_get_arg( 'relatedposts_exclude' );
-			$this->_action_frontend_init_ajax( $excludes );
+			$this->action_frontend_init_ajax( $excludes );
 		} else {
 			if ( isset( $_GET['relatedposts_hit'], $_GET['relatedposts_origin'], $_GET['relatedposts_position'] ) ) {
-				$this->_log_click( $_GET['relatedposts_origin'], get_the_ID(), $_GET['relatedposts_position'] );
-				$this->_previous_post_id = (int) $_GET['relatedposts_origin'];
+				$this->log_click( $_GET['relatedposts_origin'], get_the_ID(), $_GET['relatedposts_position'] );
```

Comment:
Looks like you'll need to address those warnings as well.

Reply:
Yeah I had ignored the nonce verification ones for a follow up since I felt like I needed to investigate them a bit more, unless you think right away we can safely ignore them.

Reply:
Looks like this is the one issue left here. Since you're touching the lines, it wants you to fix all of the problems at once.

If you add an ignore temporarily, don't forget to resolve it later. IMO the question to answer to decide whether we can ignore them permanently is "does the logic here do anything that a malicious actor could take advantage of by tricking a site admin or someone into hitting the URL?"

Reply:
I ignored the ones I think we can safely ignore, but some of these need some kind of verification. I'm not sure what the best approach is, since the nonce is never declared. 

What do you think like an approach I used with [this other PR](https://github.com/Automattic/jetpack/pull/22415) and used `filter_var` to clean things up?

Reply:
`filter_var` and nonce checking protect against different things, remember.